File: app/src/main/java/cz/marvincz/rssnotifier/xml/ABPZonedDateTimeConverter.java
package cz.marvincz.rssnotifier.xml;

import org.apache.commons.lang3.reflect.TypeLiteral;
import org.apache.commons.lang3.reflect.Typed;
import org.threeten.bp.ZonedDateTime;
import org.threeten.bp.format.DateTimeFormatter;

import java.util.Locale;

import androidx.annotation.Nullable;
import cz.marvincz.xmlpullparserconverter.AbstractXmlStringConverter;

public class ABPZonedDateTimeConverter extends AbstractXmlStringConverter<ZonedDateTime> {
    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss z", Locale.US);

    @Override
    public Typed<ZonedDateTime> getType() {
        return new TypeLiteral<ZonedDateTime>() {};
    }

    @Nullable
    @Override
    protected ZonedDateTime convertString(@Nullable String stringValue) {
        if (stringValue != null) {
            return ZonedDateTime.from(formatter.parse(stringValue));
        } else {
            return null;
        }
    }
}

File: app/src/main/java/cz/marvincz/rssnotifier/xml/ABPZonedDateTimeConverter.kt
package cz.marvincz.rssnotifier.xml

import cz.marvincz.xmlpullparserconverter.AbstractXmlStringConverter
import org.apache.commons.lang3.reflect.TypeLiteral
import org.apache.commons.lang3.reflect.Typed
import org.threeten.bp.ZonedDateTime
import org.threeten.bp.format.DateTimeFormatter
import java.util.*

class ABPZonedDateTimeConverter : AbstractXmlStringConverter<ZonedDateTime>() {

    override fun getType(): Typed<ZonedDateTime> {
        return object : TypeLiteral<ZonedDateTime>() {

        }
    }

    override fun convertString(stringValue: String?): ZonedDateTime? {
        return if (stringValue != null) {
            ZonedDateTime.from(formatter.parse(stringValue))
        } else {
            null
        }
    }

    companion object {
        private val formatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss z", Locale.US)
    }
}
